lib/Connector/OpenWeatherMapConnector.php

```diff
@@ -664,30 +667,214 @@ public function onScheduleCriteriaRequest(ScheduleCriteriaRequestInterface $even
+                ->addMetric('temperature_imperial', __('Temperature (Imperial)'))
+                    ->addValues('number', [])
+                ->addMetric('temperature_metric', __('Temperature (Metric)'))
+                    ->addValues('number', [])
+                ->addMetric('apparent_temperature_imperial', __('Apparent Temperature (Imperial)'))
+                    ->addValues('number', [])
+                ->addMetric('apparent_temperature_metric', __('Apparent Temperature (Metric)'))
                     ->addValues('number', [])
                 ->addMetric('wind_speed', __('Wind Speed'))
                     ->addValues('number', [])
                 ->addMetric('wind_direction', __('Wind Direction'))
+                    ->addValues('dropdown', [
+                        'N' => __('North'),
+                        'NNE ' => __('North-Northeast'),
+                        'NE' => __('Northeast'),
+                        'ENE' => __('East-Northeast'),
+                        'E' => __('East'),
+                        'ESE' => __('East-Southeast'),
+                        'SE' => __('Southeast'),
+                        'SSE' => __('South-Southeast'),
+                        'S' => __('South'),
+                        'SSW' => __('South-Southwest'),
+                        'SW' => __('Southwest'),
+                        'WSW' => __('West-Southwest'),
+                        'W' => __('West'),
+                        'WNW' => __('West-Northwest'),
+                        'NW' => __('Northwest'),
+                        'NNW' => __('North-Northwest'),
+                    ])
+                ->addMetric('wind_bearing', __('Wind Bearing'))
                     ->addValues('number', [])
-                ->addMetric('humidity', __('Humidity'))
+                ->addMetric('humidity', __('Humidity (Percent)'))
                     ->addValues('number', [])
                 ->addMetric('pressure', __('Pressure'))
                     ->addValues('number', [])
                 ->addMetric('visibility', __('Visibility'))
                     ->addValues('number', []);
     }
+
+    /**
+     * @param $item
+     * @param $unit
+     * @param $requestUnit
+     * @return array
+     */
+    private function processXmdsWeatherData($item, $unit, $requestUnit): array
+    {
+        $windSpeedUnit = $unit['windUnit'] ?? 'KPH';
+        $visibilityDistanceUnit = $unit['visibilityUnit'] ?? 'km';
+
+        // var to store output/response
+        $data = array();
+
+        // format the weather condition
+        $data['weather_condition'] = str_replace(' ', '_', strtolower($item['weather'][0]['main']));
+
+        // Temperature
+        // imperial = F
+        // metric = C
+        $data['temperature_imperial'] = $item['temp'];
+        $data['apparent_temperature_imperial'] = $item['feels_like'];
+
+        // Convert F to C
+        $data['temperature_metric'] = ($data['temperature_imperial'] - 32) * 5 / 9;
```

Comment:
Should we consider some rounding here? We do some rounding elsewhere, so perhaps it would make sense here too?